jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -906,7 +918,36 @@ public MethodHandle findStaticSetter(Class<?> clazz, String fieldName, Class<?>
 			/*[IF ]*/
 			/* If the new lookup class differs from the old one, protected members will not be accessible by virtue of inheritance. (Protected members may continue to be accessible because of package sharing.) */
 			/*[ENDIF]*/
+			/*[IF !Sidecar19-SE-B175]
 			int newAccessMode = accessMode & ~PROTECTED;
+			/*[ELSE]*/
+			/* The UNCONDITIONAL bit is discarded if the new lookup class differs from the old one in Java 9 */
+			int newAccessMode = accessMode & ~UNCONDITIONAL;
+			
+			/* There are 3 cases to be addressed for the new lookup class from a different module:
+			 * 1) There is no access if the package containing the new lookup class is not exported to 
+			 *    the package containing the old one.
+			 * 2) There is no access if the old lookup class is in a named module
+			 *    Note: The public access will be reserved if the old lookup class is a public lookup.
+			 * 3) The MODULE access is removed if the old lookup class is in an unnamed module.
+			 */
+			Module accessClassModule = accessClass.getModule();
+			Module lookupClassModule = lookupClass.getModule();
+			
+			if (!Objects.equals(accessClassModule, lookupClassModule)) {
+				if (!lookupClassModule.isExported(lookupClass.getPackageName(), accessClassModule)) {
+					newAccessMode = NO_ACCESS;
+				} else if (accessClassModule.isNamed()) {
+					if ((PUBLIC | UNCONDITIONAL) == accessMode) {
```

Comment:
This case is for dealing with publicLookup I presume?  Can you add a comment here to indicate that?

Also, can any other lookup object have the PUBLIC | UNCONDITIONAL flags set?  Does it also need a check for Object as the lookup class?

Reply:
publicLookup is a singleton - if this is checking public lookup, can it do an identity check against the `PUBLIC_LOOKUP` object?

Reply:
Comment added and code updated to directly check against PUBLIC_LOOKUP object (already verified in tests)